prime-router/src/main/kotlin/azure/TestDataFunction.kt

```diff
@@ -0,0 +1,181 @@
+    ): HttpResponseMessage {
+        return try {
+            val reportFile = workflowEngine.db.fetchReportFile(reportId)
+            // check format
+            if (reportFile.bodyFormat != "INTERNAL") {
+                val msg = "Report ID is not for an internal format. Will not continue."
+                context.logger.warning(msg)
+                return HttpUtilities.badRequestResponse(request, msg)
+            }
+            // check sender
+            if (reportFile.sendingOrg.isNullOrEmpty()) {
+                val msg = "Will not extract data from a child report. Sending org should not be null, but it is."
+                context.logger.warning(msg)
+                return HttpUtilities.badRequestResponse(request, msg)
+            }
+            val schema = workflowEngine.metadata.findSchema(reportFile.schemaName)
+                ?: return HttpUtilities.badRequestResponse(
+                    request,
+                    "Cannot find schema for ${reportFile.schemaName} in metadata."
+                )
+            context.logger.info("Trying to download the report")
+            val report = getReport(reportFile, schema)
+            context.logger.info("Extracting the deidentified data")
+            val deidentifiedData = report.getDeidentifiedTestData()
+            context.logger.info("Extracted ${deidentifiedData.count()} rows to insert")
+            context.logger.info("Removing old test data records for report id $reportId")
+            workflowEngine.db.transact { txn ->
+                workflowEngine.db.deleteTestDataForReportId(reportId, txn)
+                context.logger.info("Inserting deidentified data")
+                workflowEngine.db.saveTestData(deidentifiedData, txn)
+                context.logger.info("Done saving data")
+            }
+            HttpUtilities.okResponse(request, writeResponseBody(reportId, deidentifiedData.count()))
+        } catch (iex: IllegalStateException) {
+            val msg = "Report does not exist for id $reportId"
+            context.logger.severe(msg)
+            context.logger.severe(iex.localizedMessage)
+            context.logger.severe(iex.stackTraceToString())
+            HttpUtilities.badRequestResponse(request, msg)
+        } catch (bex: BlobStorageException) {
+            val msg = "Unable to download report blob for $reportId"
+            context.logger.severe(msg)
+            context.logger.severe(bex.localizedMessage)
+            context.logger.severe(bex.stackTraceToString())
+            HttpUtilities.badRequestResponse(request, msg)
+        }
+    }
+
+    private fun getReport(
+        reportFile: ReportFile,
+        schema: Schema
+    ): Report {
+        val bytes = workflowEngine.blob.downloadBlob(reportFile.bodyUrl)
+        return workflowEngine.csvSerializer.readInternal(
+            schema.name,
+            ByteArrayInputStream(bytes),
+            emptyList(),
+            blobReportId = reportFile.reportId
+        )
+    }
```

Comment:
Extracting the report here. I'm not doing any error handling because the error handling is happening higher up where it's called. I'm letting this throw if there's a problem.